src/main/java/org/datadog/jenkins/plugins/datadog/util/AsyncWriter.java

```diff
@@ -27,6 +34,12 @@ public final class AsyncWriter<T> {
     private final int stopTimeoutSeconds;
     private final int batchSizeLimit;
 
+    private final Timer submit;
+    private final Meter submitDropped;
+    private final Timer dispatch;
+    private final Gauge<Integer> queueSize;
+    private final Histogram batchSize;
```

Comment:
This is for internal usage right? We're not reporting these metrics to Datadog.

Reply:
Correct, this only shows up in diagnostic traces for now. It'd be nice to integrate this with actual telemetry, but this is too much of an effort for now.